src/ui/public/doc_table/doc_table.js

```diff
@@ -42,24 +42,6 @@ uiModules.get('kibana')
           columns: $scope.columns
         };
 
-        const prereq = (function () {
```

Comment:
The sole purpose of this piece of art (in the way it was currently used) was to wrap the below `$scope.$watch('searchSource')` callback function, and trigger the `ready:vis` event only the first time a `searchSource` would change.

Instead of placing the `courier.fetch()` call directly in this place, I placed it directly in the watch and removed this `prereq` function. The only place where `searchSource` is used in the `doc_table` is from within the dashboard embeddable. That will never change the `searchSource`, thus this callback will only be called a single time anyway. Even if it would be called multiple times, queuing another `courier.fetch()` isn't anything very bad to happen, so I removed this whole method in replace for a simple `courier.fetch()` call in the watch below.